primitives/sidechain-mc-hash/src/lib.rs

```diff
@@ -190,7 +190,11 @@ impl InherentDigest for McHashInherentDigest {
 		for item in digest {
 			if let DigestItem::PreRuntime(id, data) = item {
 				if *id == MC_HASH_DIGEST_ID {
-					return Ok(McBlockHash(data.clone()));
+					let data = data
+						.clone()
+						.try_into()
+						.map_err(|_| format!("Invalid MC hash in digest: {data:?}"))?;
```

Comment:
This time it looks like this (if digest is wrong):
`Invalid MC hash in digest: [91, 223, 29, 17, 0, 0, 0, 0]`. Why not hex, like we should do for all bytes printing?

Reply:
I think it's also a programmatic error, to have something else than 32 bytes here. It is not exceptional situation. It would be nice to add that `mcsh` has to contain 32 bytes and block creator put syntactically wrong value there.